### PR TITLE
Add BitToggle Component #262

### PR DESCRIPTION
**Pull request by hamedhajiloo**

### CODE REVIEW

File: src/Bit.Client.Web.BlazorUI.sln [hunk 1]
```diff
@@ -31,7 +31,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Bit.Client.Web.BlazorUI.Tes
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Tooling", "Tooling", "{778B3D69-5A32-453B-9A76-167912F438DE}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Bit.Tooling.SourceGenerators", "Tooling\Bit.Tooling.SourceGenerators\Bit.Tooling.SourceGenerators.csproj", "{A09BBB64-CD8C-4993-BA06-427A4C091A32}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Bit.Tooling.SourceGenerators", "Tooling\Bit.Tooling.SourceGenerators\Bit.Tooling.SourceGenerators.csproj", "{A09BBB64-CD8C-4993-BA06-427A4C091A32}"
```
Comment (msynk) [inline src/Bit.Client.Web.BlazorUI.sln:34]: please undo these changes

File: src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Toggles/BitToggleTest.razor [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿<CascadingValue Value="Visual">
+	<BitToggle IsEnabled="@IsEnabled" IsChecked="@IsChecked" IsInlineLabel="@IsInlineLabel" OnText="@OnText" OffText="@OffText"></BitToggle>
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Toggles/BitToggleTest.razor:2]: all 2-way bound parameters should be used with "@bind-" unless there is a certain reason for not doing that.

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Toggles/BitToggle.razor.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            string.IsNullOrWhiteSpace(OnText) || string.IsNullOrWhiteSpace(OffText) ?
+                $"{RootElementClass}-without-onoff-{VisualClassRegistrar()}" :
+                string.Empty);
+        }
+
+        protected virtual async Task HandleOnClick(MouseEventArgs e)
+        {
+            if (IsEnabled)
+            {
+                IsChecked = !IsChecked;
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Toggles/BitToggle.razor.cs:56]: before this change in IsChecked, checking that IsCheckChanged has a value is mandatory.
please check the CheckBox component for further information about the 2-way bound parameters.

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Toggles/BitToggle.razor.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            {
+                var isCheckedClass = IsChecked ? "checked" : "unchecked";
+                var isEnabledClass = IsEnabled ? "enabled" : "disabled";
+                return $"{RootElementClass}-{isEnabledClass}-{isCheckedClass}-{VisualClassRegistrar()}";
+            });
+
+            ClassBuilder.Register(() => IsInlineLabel ? $"{RootElementClass}-inline-{VisualClassRegistrar()}" : string.Empty);
+            ClassBuilder.Register(() =>
+            string.IsNullOrWhiteSpace(OnText) || string.IsNullOrWhiteSpace(OffText) ?
+                $"{RootElementClass}-without-onoff-{VisualClassRegistrar()}" :
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Toggles/BitToggle.razor.cs:48]: the phrase "without-onoff" could be more simplified to "noonoff" in order to reduce the length of the CSS class name.

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor [hunk 5]
```diff
@@ -276,3 +276,33 @@
         <BitChoiceOption Text="Disabled option" Value="3" IsEnabled="false"></BitChoiceOption>
     </BitChoiceGroup>
 </div>
+<hr />
+
+<div class="toggle-container">
+ <h2>BitToggle</h2>
+ <h3>Basics</h3>
+ <BitToggle @bind-IsChecked="IsToggleChecked" IsEnabled="true" OnText="On" OffText="Off">Enabled and checked</BitToggle>
```
Comment (msynk) [inline src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor:284]: did you consider the 2-way bound parameter (IsChecked) in all of these samples?

**Review by msynk — CHANGES_REQUESTED** [review]

File: src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Toggles/BitToggleTest.razor (hunk 2 above)
Comment (hamedhajiloo) [reply src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Toggles/BitToggleTest.razor:2]: Inside toggle component `IsChecked` only changes and other properties not change.
I realized that only the properties that change inside the component should be 2-way bound.

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor (hunk 5 above)
Comment (hamedhajiloo) [reply src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor:284]: Not. I do this

**Review by msynk — CHANGES_REQUESTED** [review]

File: src/Bit.Client.Web.BlazorUI.sln [hunk 6]
```diff
@@ -1,4 +1,3 @@
-﻿
```
Comment (msynk) [inline src/Bit.Client.Web.BlazorUI.sln:1]: again, undo this change 😁

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Toggles/BitToggle.razor.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                ClassBuilder.Reset();
+                _ = IsCheckedChanged.InvokeAsync(value);
+            }
+        }
+
+        [Parameter] public EventCallback<bool> IsCheckedChanged { get; set; }
+
+
+        [Parameter] public RenderFragment ChildContent { get; set; }
+        [Parameter] public EventCallback<MouseEventArgs> OnChange { get; set; }
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Toggles/BitToggle.razor.cs:32]: The OnChange event argument's type should not be "MouseEventArgs", but the type of the value that is changing itself, which is "bool" in this case.

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Toggles/BitToggle.razor.cs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+            string.IsNullOrWhiteSpace(OnText) || string.IsNullOrWhiteSpace(OffText) ?
+                $"{RootElementClass}-noonoff-{VisualClassRegistrar()}" :
+                string.Empty);
+        }
+
+        protected virtual async Task HandleOnClick(MouseEventArgs e)
+        {
+            if (IsEnabled is false || IsCheckedChanged.HasDelegate is false) return;
+            IsChecked = !IsChecked;
+            await OnChange.InvokeAsync(e);
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Toggles/BitToggle.razor.cs:56]: The argument that is passing to the OnChange event should be the changed value, which is the new value for IsChecked in here.

**Review by msynk — CHANGES_REQUESTED** [review]

**Comment by msynk** [comment]
This PR has been obsoleted with the new PR #370
